### PR TITLE
Feat/#8rdhyr #9ujbzp

### PR DESCRIPTION
**Pull request by asianswagstorm**

### CODE REVIEW

**Comment by dannylekim** [comment]
Task linked: [CU-9ujbzp Bill display update 2](https://app.clickup.com/t/9ujbzp)

**Comment by dannylekim** [comment]
Task linked: [CU-8rdhyr Home Dashboard v2](https://app.clickup.com/t/8rdhyr)

File: src/views/BillDisplay/BillDisplay.jsx [hunk 1]
```diff
@@ -3,135 +3,444 @@ import "./styles.scss";
...
+                onClick: () => this.setState({filter: {
+                                                        ...this.state.filter,
+                                                        statusOpened: true,
+                                                        categoryOpened:false,
+                                                        dateOpened: false
+                                                    },
+                                                    dateFilters: {...this.state.dateFilters, startDate: {...this.state.dateFilters.startDate, selected: false}, endDate: {...this.state.dateFilters.endDate,selected: false} }})
+            },
+            {   title: "A to Z",
+                onClick: () => this.updateBills("A to Z")
```
Comment (alphakennyn) [inline src/views/BillDisplay/BillDisplay.jsx]: can you use some `id` to determine sort type?

File: src/utils/requests/BillRequests.js [hunk 2]
```diff
@@ -3,10 +3,10 @@ const {URL} = require("../../config");
  * @function getBill
  * @description Function that calls the /bills route to return the bills associated to the user. With bearer token.
  */
-export const getBill = async () => {
+export const getBill = async (query_params) => {
```
Comment (alphakennyn) [inline src/utils/requests/BillRequests.js]: use default params, so you don't need to define param

```suggestion
export const getBill = async (query_params = '') => {
```

File: src/views/BillDisplay/BillDisplay.jsx [hunk 3]
```diff
@@ -3,135 +3,444 @@ import "./styles.scss";
...
+            }> <FaBars size={24}/><span className="filter__header__title">Filters</span> 
         </div>
-    
+        <div className="filter__selections__container">
+            {
+                filterBadges.map((badgeObject, key) =>  <Badge key={key} outline pill onClick={badgeObject.onClick}>{badgeObject.title}</Badge> )
+            }
+        </div>
+        {  (this.state.filter.dateOpened  || this.state.filter.statusOpened === true || this.state.filter.categoryOpened === true) &&
+        <div className="advanced__filter__selection">
```
Comment (alphakennyn) [inline src/views/BillDisplay/BillDisplay.jsx]: WIth just a quick glance, you can tell this code is really hectic to read. Why don't you separate this into another component

Comment (alphakennyn) [reply src/views/BillDisplay/BillDisplay.jsx]: Ill leave the naming up to you

**Review by alphakennyn — CHANGES_REQUESTED** [review]

**Comment by dannylekim** [comment]
image

File: src/components/BillFilter/BillFilter.jsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import "./styles.scss";
+import { Badge, FormCheckbox,FormInput  } from "shards-react";
+import { FaBars } from "react-icons/fa";
+
+/**
+    * @description the filter box when three horizontal lines clicked
+    */
+const BillFilter = ({dateFilters,filter,billStatusFilter, handleDateSelection,updateBills,setState}) => {
```
Comment (alphakennyn) [inline src/components/BillFilter/BillFilter.jsx]: very nice! Can you please format your code?

File: src/components/BillFilter/BillFilter.jsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            }
+          })
+        }> <FaBars size={24}/><span className="filter__header__title">Filters</span> 
+    </div>
+    <div className="filter__selections__container">
+        {
+            filterBadges.map((badgeObject, key) =>  <Badge key={key} outline pill id={badgeObject.id} onClick={badgeObject.onClick}>{badgeObject.title}</Badge>)
+        }
+    </div>
+    {  (filter.dateOpened === true  || filter.statusOpened === true || filter.categoryOpened === true) &&
```
Comment (alphakennyn) [inline src/components/BillFilter/BillFilter.jsx]: can you assign this condition to a some variable? Cleaner that way

**Review by alphakennyn — CHANGES_REQUESTED** [review]

**Review by alphakennyn — APPROVED** [review]

**Pull request merged**